mat64/dense.go

```diff
@@ -427,3 +429,63 @@ func (m *Dense) Augment(a, b Matrix) {
 	w := m.View(0, ac, br, bc).(*Dense)
 	w.Copy(b)
 }
+
+// MarshalBinary encodes the receiver into a binary form and returns the result.
+func (m Dense) MarshalBinary() ([]byte, error) {
+	var err error
+	buf := new(bytes.Buffer)
+	err = writeBin(buf, int64(m.mat.Rows))
+	if err != nil {
+		return nil, err
+	}
+
+	err = writeBin(buf, int64(m.mat.Cols))
+	if err != nil {
+		return nil, err
+	}
+
+	for i := 0; i < m.mat.Rows; i++ {
+		// We don't write out the length of the slice as
+		// it should be m.mat.Cols.
+		err = writeSlice(buf, m.rowView(i))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return buf.Bytes(), err
+}
+
+// UnmarshalBinary decodes the binary form into the receiver.
+func (m *Dense) UnmarshalBinary(data []byte) error {
+	buf := bytes.NewReader(data)
+
+	var rows int64
+	err := readBin(buf, &rows)
+	if err != nil {
+		return err
+	}
+
+	var cols int64
+	err = readBin(buf, &cols)
+	if err != nil {
+		return err
+	}
+
+	m.mat.Rows = int(rows)
+	m.mat.Cols = int(cols)
+	m.mat.Stride = int(cols)
+	m.mat.Data = make([]float64, 0, m.mat.Rows*m.mat.Cols)
```

Comment:
Now that #82 is in, you need to set m.capRows = rows and m.capCols = cols.

You'll need to rebase to see those.